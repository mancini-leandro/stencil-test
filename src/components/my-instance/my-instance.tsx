import { Component, Host, h, Prop } from '@stencil/core';

@Component({
  tag: 'my-instance',
  styleUrl: 'my-instance.css',
  shadow: true,
})
export class MyInstance {
  /**
   * prop name
   */
  @Prop()
  name: string;

  /**
   * prop lastName
   */
  @Prop()
  lastName: string;

  format(): string {
    return [this.name, this.lastName].join(' ');
  }

  render() {
    return (
      <Host>
        <p>Meu nome Ã©: {this.format()}</p>
      </Host>
    );
  }

}
