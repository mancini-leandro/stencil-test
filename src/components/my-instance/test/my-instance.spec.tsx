import { newSpecPage } from '@stencil/core/testing';
import { MyInstance } from '../my-instance';

describe('my-instance', () => {
  const html = `
    <my-instance name="Leandro" last-name="Mancini">
      <p class="cool">Est√° frio</p>
    </my-instance>
  `;

  it('should format the name', () => {
    let comp = new MyInstance();

    comp.name = 'Leandro';
    comp.lastName = 'Mancini';

    expect(comp.format()).toEqual('Leandro Mancini');
  });

  it('should render with shadow dom', async () => {
    const page = await newSpecPage({
      html,
      components: [MyInstance]
    });

    expect(page.root.shadowRoot).toBeTruthy();
    expect(page.root.querySelector('.name')).toBeFalsy();
    expect(page.root.shadowRoot.querySelector('.name')).toBeTruthy();
    expect(page.root.querySelector('.cool')).toBeTruthy();
    expect(page.root).toMatchSnapshot();
  });

  it('should render without shadow dom', async () => {
    const page = await newSpecPage({
      html,
      components: [MyInstance],
      supportsShadowDom: false
    });

    expect(page.root.shadowRoot).toBeFalsy();
    expect(page.root.querySelector('.name')).toBeTruthy();
    expect(page.root.querySelector('.cool')).toBeTruthy();
  });
});
