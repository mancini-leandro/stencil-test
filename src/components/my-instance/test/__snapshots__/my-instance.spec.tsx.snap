// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-instance should render with shadow dom 1`] = `
<my-instance last-name="Mancini" name="Leandro">
  <mock:shadow-root>
    <p class="name">
      Meu nome é: Leandro Mancini
    </p>
    <slot></slot>
  </mock:shadow-root>
  <p class="cool">
    Está frio
  </p>
</my-instance>
`;
